docs/commands.md

```diff
@@ -7,81 +9,92 @@
+
+The `-f` flag can be used multiple times to add different files or directories to the bundle.
 
 ### Generating a [BundleLock](resources.md#bundlelock)
 
-`push` may also output a BundleLock file for users that would like a deterministic reference to a pushed bundle. For example, running:
+`push` command can output a [`BundleLock` configuration](resources.md#bundlelock-configuration) for users that would like a deterministic reference to a pushed bundle. For example, running:
 
-`$ impgpkg push -f my-bundle -b index.docker.io/k8slt/sample-bundle:v0.1.0 --lock-output
-bundle.lock.yml`
+```bash
+$ impgpkg push -b index.docker.io/k8slt/sample-bundle:v0.1.0 -f my-bundle/ --lock-output
+/tmp/bundle.lock.yml
+```
+
+will create `/tmp/bundle.lock.yml` with BundleLock configuration. If another bundle image in the repository is later given the same tag (`v0.1.0`), the BundleLock configuration will continue to provide immutable reference (via digest) to the original pushed bundle.
 
-will output a [BundleLock](resources.md#bundlelock) file to `bundle.lock.yml`. If another bundle image in the repository is later given the same tag (`v0.1.0`), the BundleLock will guarantee users continue to reference the original bundle by its digest.
+---
 ## Pull
 
-### Pulling an artifact
+### Overview
 
 After pushing bundles to a registry, users can retrieve them with `imgpkg pull`. For example,
 
-`$ imgpkg pull -b index.docker.io/k8slt/sample-bundle -o my-bundle`
+```bash
+$ imgpkg pull -b index.docker.io/k8slt/sample-bundle -o my-bundle/
+```
 
-will pull a bundle from `index.docker.io/k8slt/sample-bundle` and extract its
-contents in to the `my-bundle` directory, which gets created if it does not
-exist.
+will pull a bundle from `index.docker.io/k8slt/sample-bundle` and extract its contents in to the `my-bundle/` directory, which gets created if it does not exist.
 
-When pulling a bundle, imgpkg must ensure that the referenced images are updated
-to account for any relocations. Because images are referenced by digest, imgpkg
-will search for all the referenced images in the same repository as the bundle.
-If all referenced digests are found, imgpkg will rewrite the bundle's
-[ImagesLock](resources.md#imageslock) with updated references, however, if any
-of the image digests are not found in the repository, imgpkg will not update the
-references.
+When pulling a bundle, imgpkg ensures that the referenced images are updated to account for any relocations. Because images are referenced by digest, imgpkg will search for all the referenced images in the same repository as the bundle. If all referenced digests are found, imgpkg will update image references in bundle's [`.imgpkg/images.yml` file](resources.md#imgpkg-directory). If any of the digests are not found in the repository, imgpkg will not update the references.
```

Comment:
```suggestion
When pulling a bundle, imgpkg ensures that the referenced images are updated to account for any relocations. It will search for each referenced image by digest in the same repository as the bundle. If all referenced digests are found, imgpkg will update image references in the bundle's [`.imgpkg/images.yml` file](resources.md#imgpkg-directory). If any of the digests are not found in the repository, imgpkg will not update any references.
```